### PR TITLE
[dv,chip] add chip_sw_ast_clk_rst_inputs test

### PR DESCRIPTION
**Pull request by ariel9678**
Signed-off-by: arielc <ariel.cohen@nuvoton.com>
This is a draft pr for issue #14132.
Please ignore lint errors and comments, this is just for reviewing the test content.

By the way when pulling the latest repository version, it fails with the following error:
_INFO: From Running Cargo build script libftdi1_sys:
Build Script Warning: Config for libftdi1 not found, falling back to default search path: `"pkg-config" "--libs" "--cflags" "libftdi1" "libftdi1 >= 1.4"` did not exit successfully: exit status: 1
[928 / 1,306] Action hw/top_earlgrey/ip/ast/data/ast_regs.h; 47s processwrapper-sandbox ... (24 actions, 23 running)
[1,024 / 1,306] Compiling Rust rlib serde v1.0.145 (21 files); 41s processwrapper-sandbox ... (24 actions running)
ERROR: /tanap1/proj_cd14/opentitan/arcohen/chip_clk_rst_pr/opentitan/sw/host/opentitanlib/bindgen/BUILD:33:21: Generating bindings for sw/host/opentitanlib/bindgen/difs.h.. failed: (Exit 101): bindgen__bin failed: error executing command bazel-out/k8-opt-exec-2B5CBBC6-ST-e846b08c7501/bin/external/rules_rust_bindgen__bindgen-0.60.1/bindgen__bin --no-rustfmt-bindings '--allowlist-type=dif_lc_ctrl_state' ... (remaining 32 arguments skipped)_

Any idea what is the reason for this error?

### CODE REVIEW

**Comment by sriyerg** [comment]
It will be great to have the testplan entry for this test updated as a part of this PR, so that the test intent / method is clear. It will also help review this PR quicker.

**Comment by ariel9678** [comment]
@sriyerg 
Testplan is updated.

File: sw/device/tests/chip_sw_ast_clk_rst_inputs.c [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+bool test_main() {
+
+  uint32_t event_idx;
+  int idx=1;
+  int read_fifo_depth_val;
+
+  bool deepsleep;
+  dif_pwrmgr_domain_config_t pwrmgr_config;
+
+    const dif_entropy_src_config_t entropy_src_config = {
```
Comment (matutem) [inline sw/device/tests/chip_sw_ast_clk_rst_inputs.c:452]: Indentation...

File: sw/device/tests/chip_sw_ast_clk_rst_inputs.c [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+              .cmd = 0x00000001 |  // Reseed from entropy source only.
+                     kMultiBitBool4False << 8,
+              .seed_material =
+                  {
+                      .len = 0,
+                  },
+          },
+      .reseed_cmd =
+          {
+              .cmd = 0x00001002 |  // One generate returns 1 block, reseed
```
Comment (matutem) [inline sw/device/tests/chip_sw_ast_clk_rst_inputs.c:422]: Is there a way to express 0x1002 with symbols to make it more obvious what this means? Same above and below.

File: sw/device/tests/chip_sw_ast_clk_rst_inputs.c [hunk 3]
```diff
@@ -0,0 +1,498 @@
+// Copyright lowRISC contributors.
```
[the remark this answers is not in the document]
Comment (tjaychen) [reply sw/device/tests/chip_sw_ast_clk_rst_inputs.c:1]: btw there's an easy way to do it through the bazel flow.

`bazel run //quality:clang_format_fix` or bazelisk.sh if you're not in an airgapped system.

**Comment by msfschaffner** [comment]
Closing this draft PR as it is superseded by #16581.

**Pull request closed without merging**